#129: Validating credentials model

mehoil opened the pull request:
closes #129

Vafeen reviewed (commented):
Sign-in returns No-Content, but must return id

Vafeen reviewed (changes requested):
Sign-in returns No-Content, but must return id 
@mehoil 

UPD: also sign-up must return No-Content in success

mehoil commented:
fixed

Vafeen commented:
#137 #138 @mehoil

Vafeen reviewed (changes requested):
In userRepositoryImpl:58 we can't save deviceId with user data. because user can have more than one device

Vafeen commented:
We should save userId in separate place with tokens, because in future i will think about feature: "to close access to account for sth device"

mehoil commented:
Extracted `device_id` to a separate new table `users_devices` (user to device link);
Updated `users.sql` => drop & restart DB container;
Required `deviceId` field upon `/api/sign-in`
Added `UsersDevices` table, DAO, repository.
